File: ssi-chat/src/chat_application/auth/subject.py
from __future__ import annotations

from pydantic import BaseModel, Field


class Subject(BaseModel):
    user_id: str
    given_name: str | None = None
    family_name: str | None = None
    title: str
    lob: str | None = None
    roles: list[str] = Field(min_length=1)
    groups: list[str] = Field(default_factory=list)
    supervisor_id: str | None = None
    covering_lobs: list[str] = Field(default_factory=list)

File: ssi-chat-j/src/main/java/com/sanjuthomas/policypilot/auth/Subject.java
package com.sanjuthomas.policypilot.auth;

import java.util.List;

public record Subject(
    String userId,
    String givenName,
    String familyName,
    String title,
    String lob,
    List<String> roles,
    List<String> groups,
    String supervisorId,
    List<String> coveringLobs,
    String bearerToken,
    String sessionId) {

  public Subject withTokens(String bearerToken, String sessionId) {
    return new Subject(
        userId,
        givenName,
        familyName,
        title,
        lob,
        roles,
        groups,
        supervisorId,
        coveringLobs,
        bearerToken,
        sessionId);
  }
}
